### PR TITLE
Move `LastProcessedPositionState` to stream processor

### PR DESCRIPTION
**Pull request by pihme**
## Description
The last processed position state is only used by the StreamProcessor. Therefore, it
was moved into the streamprocessor package. The existing classes and interfaces were
moved as is. There is probably a way to simplify this code, but this is something we
can still do after the refactoring.

In the tests, minimal effort was undertaken to make the tests pass. The test classes
are very complex and ripe for an overhaul, but this would have increased the change
set considerably.

## Related issues

related to #9725

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java [hunk 1]
```diff
@@ -147,6 +150,11 @@ public static StreamProcessorBuilder builder() {
     return new StreamProcessorBuilder();
   }
 
+  @Deprecated // only used for tests
```
Comment (npepinpe) [inline engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java:153]: :upside_down_face: We're starting to see this more and more, might be time to introduce our own annotation, much like Google's `@VisibleForTesting` or the likes. This is a very soft suggestion :)

**Review by npepinpe — APPROVED** [review]
:rocket: 

- :+1: Love seeing the state interface get smaller/split up

Just a small suggestion, mostly for future things.

File: engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java (hunk 1 above)
Comment (pihme) [reply engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java:153]: I'll keep it in the back of my head. Still hoping that the methods can soon be removed during refactoring, but I agree it's on the threshold to think about it.

**Comment by pihme** [comment]
bors merge

**Comment by pihme** [comment]
bors retry

**Pull request merged**